src/metabase/analytics/snowplow.clj

```diff
@@ -0,0 +1,172 @@
+  :visibility :public
+  :setter     :none
+  :getter     #(public-settings/uuid-nonce :analytics-uuid))
+
+(defsetting snowplow-available
+  (str (deferred-tru "Boolean indicating whether a Snowplow collector is available to receive analytics events.")
+       " "
+       (deferred-tru "Should be set via environment variable in Cypress tests or during local development."))
+  :type       :boolean
+  :visibility :public
+  :default    config/is-prod?)
+
+(defsetting snowplow-enabled
+  (str (deferred-tru "Boolean indicating whether analytics events are being sent to Snowplow. True if anonymous tracking")
+       " "
+       (deferred-tru "is enabled for this instance, and a Snowplow collector is available."))
+  :type   :boolean
+  :setter :none
+  :getter (fn [] (and (snowplow-available)
+                      (public-settings/anon-tracking-enabled)))
+  :visibility :public)
+
+(defsetting snowplow-url
+  (deferred-tru "The URL of the Snowplow collector to send analytics events to.")
+  :default    (if config/is-prod?
+                "https://sp.metabase.com"
+                ;; See the iglu-schema-registry repo for instructions on how to run Snowplow Micro locally for development
+                "http://localhost:9090")
+  :visibility :public)
+
+(def ^:private emitter
+  "Returns an instance of a Snowplow emitter"
+  (let [emitter* (delay
+                   (let [client (-> (HttpClients/custom)
+                                    (.setConnectionManager (PoolingHttpClientConnectionManager.))
+                                    (.build))
+                         builder (-> (ApacheHttpClientAdapter/builder)
+                                     (.httpClient client)
+                                     (.url (snowplow-url)))
+                         adapter (.build ^ApacheHttpClientAdapter$Builder builder)
+                         batch-emitter-builder (-> (BatchEmitter/builder)
+                                                   (.bufferSize 1)
+                                                   (.httpClientAdapter adapter))]
+                     (.build ^BatchEmitter$Builder batch-emitter-builder)))]
+     (fn [] @emitter*)))
+
+(def ^:private tracker
+  "Returns instance of a Snowplow tracker"
+  (let [tracker* (delay
+                  (-> (Tracker$TrackerBuilder. ^Emitter (emitter) "sp" "metabase")
+                      .build))]
+    (fn [] @tracker*)))
+
+(defn- set-subject
+  "Create a Subject object for a given user ID, to be included in analytics events"
+  [builder user-id]
+  (if user-id
+    (let [subject (-> (Subject$SubjectBuilder.)
+                      (.userId (str user-id))
+                      .build)]
+      (.subject ^Unstructured$Builder builder subject))
+    builder))
+
+(defn- context
+  "Common context included in every analytics event"
+  []
+  (new SelfDescribingJson
+       "iglu:com.metabase/instance/jsonschema/1-0-0"
```

Comment:
is this 1-0-0 related to the other 1-0-0s? If so, which of the four (account, invite, etc) does it come from?

Reply:
This is a separate `instance` schema that is defines the data sent with every event. It should be included in the `event->schema` map as well; good call.